environs/bootstrap/bootstrap.go

```diff
@@ -659,15 +659,33 @@ func isCompatibleVersion(v1, v2 version.Number) bool {
 // for tools syncing, and adds an image metadata source after verifying
 // the contents.
 func setPrivateMetadataSources(metadataDir string) ([]*imagemetadata.ImageMetadata, error) {
-	logger.Infof("Setting default tools and image metadata sources: %s", metadataDir)
-	tools.DefaultBaseURL = metadataDir
+	if _, err := os.Stat(metadataDir); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, errors.Annotate(err, "cannot access metadata directory")
+		}
+		logger.Infof("metadata source: %s, not found", metadataDir)
+		return nil, nil
+	}
+
+	toolsMetadataDir := filepath.Join(metadataDir, storage.BaseToolsPath)
+	if _, err := os.Stat(toolsMetadataDir); err != nil {
+		if !os.IsNotExist(err) {
+			return nil, errors.Annotate(err, "cannot access tools metadata")
+		}
+		logger.Infof("default tools metadata source: %s", tools.DefaultBaseURL)
+	} else {
+		logger.Infof("setting default tools metadata source: %s", toolsMetadataDir)
+		tools.DefaultBaseURL = metadataDir
+	}
 
 	imageMetadataDir := filepath.Join(metadataDir, storage.BaseImagesPath)
```

Comment:
we want the same logic as for tools above.
ie
the user can specify --metadata-source as one of:

somedir
somedir/tools
somedir/images

and the logic should Do The Right Thing